src/main/kotlin/se/uulm/snowballr/backend/service/ProjectService.kt

```diff
@@ -369,4 +395,41 @@ class ProjectService(
             }
         return getProjectPapers(request, predicate)
     }
+
+    override suspend fun addPaperToProject(request: GrpcProjectPaper.Add): GrpcProjectPaper {
+        val currentUser = userRepo.getUserById(GrpcContext.getUserIdFromContext())
+        val projectId = parseUUID(request.projectId, EntityType.PROJECT)
+
+        if (!isProjectMember(projectId, currentUser.id)) {
+            verifyServerAdminRole(currentUser) {
+                throw UnauthorizedException.Single(
+                    EntityType.PROJECT,
+                    projectId.toString(),
+                    AccessType.READ,
+                    it,
+                )
+            }
+        }
+
+        val paperId = parseUUID(request.paperId, EntityType.PAPER)
+        val project = repo.getProjectById(projectId)
+        val paper = paperRepo.getPaperById(paperId)
+        when {
+            projectPaperRepo.doesProjectPaperExistsByPaperId(paperId) -> {
+                throw SnowballRException.DuplicateEntityException(
+                    EntityType.PROJECT_PAPER,
+                    projectId.toString(),
+                    paperId.toString(),
+                )
+            }
+        }
```

Comment:
This can be simplified and the indentation reduced.
```suggestion

        if (projectPaperRepo.doesProjectPaperExistsByPaperId(paperId)) {
            throw SnowballRException.DuplicateEntityException(
                EntityType.PROJECT_PAPER,
                projectId.toString(),
                paperId.toString(),
            )
        }
```

Reply:
Good point.